index.js

```diff
@@ -15,23 +15,27 @@ export function includeKeys(object, predicate) {
 			}
 		}
 	} else {
-		// `for ... of Object.keys()` is faster than `for ... of Object.entries()`.
-		for (const key of Object.keys(object)) {
-			const value = object[key];
-			if (predicate(key, value, object)) {
-				Object.defineProperty(result, key, {
-					value,
-					writable: true,
-					enumerable: true,
-					configurable: true,
-				});
+		// `for ... of Reflect.ownKeys()` is faster than `for ... of Object.entries()`.
+		for (const key of Reflect.ownKeys(object)) {
+			if (isEnumerable.call(object, key)) {
+				const value = object[key];
+				if (predicate(key, value, object)) {
+					Object.defineProperty(result, key, {
+						value,
+						writable: true,
+						enumerable: true,
+						configurable: true,
+					});
+				}
 			}
 		}
 	}
 
 	return result;
 }
 
+const {propertyIsEnumerable: isEnumerable} = Object.prototype;
```

Comment:
`propertyIsEnumerable` is also a global function, so the property is renamed to avoid shadowing.